src/Sarif.Multitool.FunctionalTests/TestData/UriBaseIdRequiresRelativeUri/Invalid.sarif

```diff
@@ -98,54 +67,89 @@
           "stdoutStderr": {
             "uri": "file:///c:/log/out-err.txt",
             "uriBaseId": "%SRCROOT%"
-          },
-          "toolNotifications": [
+          }
+        }
+      ],
+      "conversion": {
+        "tool": {
+          "name": "ConversionTool"
+        },
+        "analysisToolLogFiles": [
+          {
+            "uri": "file:///C:/src/file.c",
+            "uriBaseId": "%SRCROOT%"
+          }
+        ]
+      },
+      "versionControlProvenance": [
+        {
+          "repositoryUri": "https://github.com/example-corp/browser",
+          "revisionId": "de67ef7",
+          "mappedTo": {
+            "uri": "file:///C:/src/file.c",
+            "uriBaseId": "%SRCROOT%"
+          }
+        }
+      ],
+      "files": [
+        {
+          "fileLocation": {
```

Comment:
>fileLocation [](start = 11, length = 12)

I had to update this entire file to SARIF v2 because otherwise the prerelease transformer would manufacture a new (correct) uri for the file. Rather than try to special-case the helper just to provoke testing of invalid content, just updated the file.

Reply:
If I understand what's going on, the problem centers on the call to `PrereleaseCompatibilityTransformer.UpdateToCurrentVersion` in `FileDiffingTests.RunTest`. This call was _fixing_ the invalid `uri` property in this `fileLocation` object. By updating this file to the most current pre-release version, you avoided the file update altogether -- and that works because `RunTest` calls `PrereleaseCompatibilityTransformer.UpdateToCurrentVersion` with `forceUpdate: false` -- right?

---
In reply to: [248137397](https://github.com/Microsoft/sarif-sdk/pull/1215#discussion_r248137397) [](ancestors = 248137397)

Reply:
Exactly right. 

---
In reply to: [248374537](https://github.com/Microsoft/sarif-sdk/pull/1215#discussion_r248374537) [](ancestors = 248374537,248137397)

Reply:
In that case I'm going to have to revisit one of the fixes in my branch. One of the unit tests of `SarifVersionOneToCurrentVisitor` was failing because the pre-release transformer was _not_ updating one of the test files -- and I fixed it in my branch by changing `forceUpdate` to `true` in `PrereleaseCompatibilityTransformer.UpdateToCurrentVersion`! (Remember I sent you a message last night saying that I'd fixed a "bug" in `PrereleaseCompatibilityTransformer.UpdateToCurrentVersion ` :-))

So I'll change it back and change the test file instead.

---
In reply to: [248377880](https://github.com/Microsoft/sarif-sdk/pull/1215#discussion_r248377880) [](ancestors = 248377880,248374537,248137397)